src/client/src/containers/AzureSubscriptions/index.tsx

```diff
@@ -110,29 +118,72 @@ class AzureSubscriptions extends React.Component<Props, IState> {
     }
     return () => {};
   }
+
+  public getServicesOrganizer(
+    type: string | undefined,
+    isLoggedIn: boolean,
+    setDetailPage: any,
+    title: any
+  ) {
+    const { formatMessage } = this.props.intl;
+    return (
+      <div
+        className={classnames(styles.servicesContainer, {
+          [styles.overlay]: !isLoggedIn
+        })}
+      >
+        <div className={styles.servicesCategory}>
+          <div className={styles.categoryTitle}>{type}</div>
+          <div className={styles.categoryDescriptor}>
+            {formatMessage(title)}
+          </div>
+          <div className={styles.servicesCategoryContainer}>
+            {azureServiceOptions.map(option => {
+              if (option.type === type) {
+                return (
+                  <div
+                    key={JSON.stringify(option.title)}
+                    className={classnames(styles.subscriptionCardContainer)}
+                  >
+                    <Card
+                      option={option}
+                      buttonText={this.addOrEditResourceText(
+                        option.internalName
+                      )}
+                      handleButtonClick={this.getServicesModalOpener(
+                        option.internalName
+                      )}
+                      disabled={!isLoggedIn}
+                      handleDetailsClick={setDetailPage}
+                      useNormalButtons={this.isSelectionCreated(
+                        option.internalName
+                      )}
+                    />
+                  </div>
+                );
+              }
+            })}
+          </div>
+        </div>
+      </div>
+    );
+  }
   public render() {
     const { isLoggedIn, setDetailPage } = this.props;
+    const serviceTypes = azureServiceOptions.map(option => option.type);
+    const uniqueServiceTypes = serviceTypes.filter(
+      (type, index, types) => types.indexOf(type) === index
+    );
     return (
       <div className={styles.container}>
-        {azureServiceOptions.map(option => (
-          <div
-            key={JSON.stringify(option.title)}
-            className={classnames(styles.subscriptionCardContainer, {
-              [styles.overlay]: !isLoggedIn
-            })}
-          >
-            <Card
-              option={option}
-              buttonText={this.addOrEditResourceText(option.internalName)}
-              handleButtonClick={this.getServicesModalOpener(
-                option.internalName
-              )}
-              disabled={!isLoggedIn}
-              handleDetailsClick={setDetailPage}
-              useNormalButtons={this.isSelectionCreated(option.internalName)}
-            />
-          </div>
-        ))}
+        {uniqueServiceTypes.map(serviceType => {
+          return this.getServicesOrganizer(
+            serviceType,
+            isLoggedIn,
+            setDetailPage,
+            messages.hostingTitle
```

Comment:
Why do you have messages.hostingTitle for all the Azure services?
`Azure Functions` should have `hostingTitle` but `CosmosDB` should have 'storageTitle`

Reply:
To fix this i would do something like this.
```ts
const serviceTypeTitle = {
  servicesEnum.DATABASE: message.storageTitle,
  servicesEnum.HOSTING: message.hostingTitle
}
```
Then pass it via:
```ts
serviceTypeTitle[serviceType]
```

(Syntax may be a bit off but that should give you an idea)

Reply:
My bad! I just changed it  now so it accounts for both cases with conditionals